File: objc/Project8/Project8/ButtonNode.m
//
//  ButtonNode.m
//  Project8
//
//  Created by Jinwoo Kim on 2/13/21.
//

#import "ButtonNode.h"

@interface ButtonNode ()
@property SKTexture *focusedImage;
@property SKTexture *unfocusedImage;
@end

@implementation ButtonNode

- (BOOL)canBecomeFocused {
    return YES;
}

- (void)setFocusedImageNamed:(NSString *)name {
    self.focusedImage = [SKTexture textureWithImageNamed:name];
    self.unfocusedImage = self.texture;
    [self setUserInteractionEnabled:YES];
}

- (void)didGainFocus {
    self.texture = self.focusedImage;
}

- (void)didLoseFocus {
    self.texture = self.unfocusedImage;
}

@end

File: swift/Project8/Project8/ButtonNode.swift
//
//  ButtonNode.swift
//  Project8
//
//  Created by Jinwoo Kim on 2/12/21.
//

import SpriteKit

class ButtonNode: SKSpriteNode {
    var focusedImage: SKTexture!
    var unfocusedImage: SKTexture!
    
    override var canBecomeFocused: Bool {
        return true
    }
    
    func setFocusedImage(named name: String) {
        focusedImage = SKTexture(imageNamed: name)
        unfocusedImage = self.texture!
        isUserInteractionEnabled = true
    }
    
    func didGainFocus() {
        texture = focusedImage
    }
    
    func didLoseFocus() {
        texture = unfocusedImage
    }
}
